apps/stats-dapp/src/containers/ProposalsTable/ProposalsTable.tsx

```diff
@@ -1,3 +1,341 @@
+  //     </Button>
+  //   ),
+  // }),
+];
+
+const PROPOSAL_TYPES: ProposalType[] = [
+  ProposalType.AnchorCreate,
+  ProposalType.AnchorUpdate,
+  ProposalType.Evm,
+  ProposalType.FeeRecipientUpdate,
+  ProposalType.MaxDepositLimitUpdate,
+  ProposalType.MinWithdrawalLimitUpdate,
+  ProposalType.Refresh,
+  ProposalType.RescueTokens,
+  ProposalType.ResourceIdUpdate,
+  ProposalType.ResourceIdUpdate,
+  ProposalType.SetTreasuryHandler,
+  ProposalType.SetVerifier,
+  ProposalType.TokenAdd,
+  ProposalType.TokenRemove,
+  ProposalType.WrappingFeeUpdate,
+];
+const PROPOSAL_STATUS: ProposalBatchStatus[] = [
+  ProposalBatchStatus.Signed,
+  ProposalBatchStatus.Removed,
+  ProposalBatchStatus.Expired,
+];
+
+function mapProposalStatusToChipColor(status: ProposalBatchStatus): ChipColors {
+  switch (status) {
+    case ProposalBatchStatus.Signed:
+      return 'green';
+    case ProposalBatchStatus.Removed:
+      return 'red';
+    case ProposalBatchStatus.Expired:
+      return 'yellow';
+  }
+
+  return 'purple';
+}
+
 export const ProposalsTable = () => {
-  return <div>Proposal Table</div>;
+  const [{ pageIndex, pageSize }, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: 10,
+  });
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex,
+      pageSize,
+    }),
+    [pageIndex, pageSize]
+  );
+
+  const chains = useMemo<Array<[string, ChainConfig]>>(
+    () =>
+      Object.keys(chainsConfig).map((key: any) => [
+        String(key),
+        chainsConfig[key],
+      ]),
+    []
+  );
+
+  const [globalFilter, setGlobalFilter] = useState('');
+  const [selectedProposalsStatuses, setSelectedProposalStatuses] = useState<
+    'all' | ProposalBatchStatus[]
+  >('all');
+  const [selectedProposalTypes, setSelectedProposalTypes] = useState<
+    'all' | ProposalType[]
+  >('all');
+  const [selectedChains, setSelectedChains] = useState<
+    'all' | [string, ChainConfig][]
+  >('all');
+
+  const pageQuery: BatchedProposalsQuery = useMemo(
+    () => ({
+      offset: pagination.pageIndex * pageSize,
+      perPage: pagination.pageSize,
+      orderBy: ProposalBatchesOrderBy.IdDesc,
+      filter: null,
+    }),
+    [
+      pageSize,
+      pagination.pageIndex,
+      pagination.pageSize,
+      selectedProposalTypes,
+      selectedProposalsStatuses,
+      selectedChains,
+    ]
+  );
+
+  const batchedProposals = useBatchedProposals(pageQuery);
+
+  const totalItems = useMemo(() => {
+    if (batchedProposals.val) {
+      return batchedProposals.val.pageInfo.count;
+    }
+    return 0;
+  }, [batchedProposals]);
+
+  const pageCount = useMemo(
+    () => Math.ceil(totalItems / pageSize),
+    [pageSize, totalItems]
+  );
+
+  const data = useMemo(() => {
+    if (batchedProposals.val) {
+      const proposalIds = batchedProposals.val.items.map((item) => item.id);
+      localStorage.setItem('proposalIds', JSON.stringify(proposalIds));
```

Comment:
I'm curious as to why we need to store this info on local storage.

Reply:
@AtelyPham Needed this in previous implementation to paginate between individual proposals when the proposal detail drawer is opened. I've removed it now. Might add it later if needed or if there's no workarounds to get the drawer pagination working.